packages/manager/apps/dedicated/client/app/account/user/components/newAccountForm/field/new-account-form-field-component.controller.js

```diff
@@ -412,7 +413,12 @@ export default class NewAccountFormFieldController {
     }
 
     if (isObject(this.rule.tracking)) {
-      this.atInternet.trackClick(this.rule.tracking);
+      this.atInternet.trackClick({
+        ...this.rule.tracking,
+        name: isFunction(this.rule.tracking.name)
+          ? this.rule.tracking.name(value)
+          : this.rule.tracking.name,
```

Comment:
**nitpick:** maintain consistency in the type of `name`. It can always be string.